tests/test_pydantic_factory.py

```diff
@@ -20,6 +20,28 @@ class AFactory(ModelFactory[A]):
         assert AFactory.build()
 
 
+def test_optional_with_constraints() -> None:
+    """this is a flaky test - because it depends on randomness, hence it's been re-ran multiple times."""
+
+    class A(BaseModel):
+        a: int | None = Field(None, ge=0, le=1)
+
+    class AFactory(ModelFactory[A]):
+        __model__ = A
+
+    has_failed = False
+    exception: Exception | None = None
+    for _ in range(100):
+        try:
+            assert isinstance(AFactory.build().a, (int, type(None)))
+        except ValidationError as e:
+            exception = e
+            has_failed = True
+
+    if has_failed:
+        pytest.fail(reason=str(exception))
```

Comment:
Not sure how these types of these should be written, let me know if that's not the best solution.

Reply:
Just a wild suggestion, maintainers can correct me if I am wrong / if there are better ways, but what about setting a seed?

```python
@pytest.mark.parametrize(["random_seed", "output_type"], [[1, int], [2, NoneType]])
def test_optional_with_constraints(random_seed: int, output_type: type) -> None:
    """this is a flaky test - because it depends on randomness, hence it's been re-ran multiple times."""

    class A(BaseModel):
        a: int | None = Field(None, ge=0, le=1)

    class AFactory(ModelFactory[A]):
        __model__ = A
        __random_seed__ = random_seed

    assert isinstance(AFactory.build().a, output_type)
    ```